### PR TITLE
Capture traces for updateParseTreeOnEdit

### PR DESCRIPTION
**Pull request by philipp-spiess**
Capture traces for `updateParseTreeOnEdit`

## Test plan

- Trivial, CI will check

### CODE REVIEW

**Comment by philipp-spiess** [comment]
@valerybugakov Saw that there's also `observableParse`. Should we use this instead? Not sure why we would need two places but I assume this place can be very high frequency (since it's on every document change)

File: vscode/src/tree-sitter/parse-tree-cache.ts [hunk 1]
```diff
@@ -73,41 +74,46 @@ function getLanguageIfTreeSitterEnabled(document: TextDocument): SupportedLangua
...
-            startPosition: startPoint,
-            oldEndPosition: oldEndPoint,
-            newEndPosition: newEndPoint,
-        })
-    }
-
-    const updatedTree = parser.safeParse(document.getText(), tree)
-    parseTreesPerFile.set(cacheKey, updatedTree)
+    wrapInActiveSpan('updateParseTreeOnEdit', span => {
+        span.setAttribute('sampled', true)
```
Comment (valerybugakov) [inline vscode/src/tree-sitter/parse-tree-cache.ts:78]: Do we want to sample all these spans? The volume will be huge.

File: vscode/src/tree-sitter/parse-tree-cache.ts [hunk 2]
```diff
@@ -73,41 +74,46 @@ function getLanguageIfTreeSitterEnabled(document: TextDocument): SupportedLangua
...
+                startIndex,
+                oldEndIndex,
+                newEndIndex,
+                startPosition: startPoint,
+                oldEndPosition: oldEndPoint,
+                newEndPosition: newEndPoint,
+            })
+        }
+
+        const updatedTree = parser.safeParse(document.getText(), tree)
```
Comment (valerybugakov) [inline vscode/src/tree-sitter/parse-tree-cache.ts:114]: If we decide to sample all spans here, we can move `span.setAttribute('sampled', true)` and `span.setAttribute('languageId', languageId)` into the `observableParse` and use it here.

**Comment by valerybugakov** [comment]
> Not sure why we would need two places, but I assume this place can be very high frequency

@philipp-spiess, that's right. I didn't instrument this part because it happens on every keystroke change.

**Review by olafurpg — APPROVED** [review]

File: vscode/src/tree-sitter/parse-tree-cache.ts (hunk 1 above)
Comment (philipp-spiess) [reply vscode/src/tree-sitter/parse-tree-cache.ts:78]: No I think that would probably be too large, I agree.

**Pull request closed without merging**